planutils/package_installation.py

```diff
@@ -0,0 +1,82 @@
+
+import json, os, glob, subprocess
+
+from planutils import settings
+
+PACKAGES = {}
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+for conf_file in glob.glob(os.path.join(CUR_DIR, 'packages', '*')):
+    base = os.path.basename(conf_file)
+    if base not in ['README.md', 'TEMPLATE']:
+        with open(os.path.join(conf_file, 'manifest.json'), 'r') as f:
+            config = json.load(f)
+        assert base not in PACKAGES, "Error: Duplicate package config -- %s" % base
+        PACKAGES[base] = config
+
+
+def check_installed(target):
+    return target in settings.load()['installed']
+
+
+def uninstall(target):
+    if 'list' == target:
+        print("\nInstalled Packages:")
+        for p in PACKAGES:
+            if check_installed(p):
+                print(" - %s: %s" % (p, PACKAGES[p]['name']))
+        print()
+        return
+    
+    if target not in PACKAGES:
+        print("Error: Package not found -- %s" % target)
+        return
+
+    if not check_installed(target):
+        print("%s isn't installed." % target)
+    else:
+        print ("Uninstalling %s..." % target)
+        s = settings.load()
+        # map a package to all those that depend on it
+        deps = {}
```

Comment:
`deps` is not a great name because `deps[p]` does not contain the dependencies of `p`. I don't have a good suggestion, though.

Unrelated, I can recommend `from collections import defaultdict` and `deps = defaultdict(set)`. That would save the `if` statement in the loop.